packages/contracts/contracts/CollateralManager.sol

```diff
@@ -0,0 +1,302 @@
+            infos_[i] = collateral.collateralInfo[collateralAddresses[i]];
+        }
+    }
+
+    function deposit(
+        uint256 _bidId,
+        ICollateralEscrowV1.Collateral[] calldata _collateral
+    ) external {
+        commitCollateral(_bidId, _collateral);
+    }
+
+    function _deposit(
+        uint256 _bidId,
+        ICollateralEscrowV1.Collateral memory collateralInfo
+    )
+        public
+        payable
+    {
+        require(collateralInfo._amount > 0, 'Collateral not validated');
+        (address escrowAddress, address borrower) = _deployEscrow(_bidId);
+        ICollateralEscrowV1 collateralEscrow = ICollateralEscrowV1(escrowAddress);
+        // Pull collateral from borrower & deposit into escrow
+        if (collateralInfo._collateralType == ICollateralEscrowV1.CollateralType.ERC20) {
+            IERC20Upgradeable(collateralInfo._collateralAddress)
+            .transferFrom(
+                borrower,
+                address(this),
+                collateralInfo._amount
+            );
+            IERC20Upgradeable(collateralInfo._collateralAddress)
+                .approve(escrowAddress, collateralInfo._amount);
+            collateralEscrow.depositToken(
+                collateralInfo._collateralAddress,
+                collateralInfo._amount
+            );
+        }
+        if (collateralInfo._collateralType == ICollateralEscrowV1.CollateralType.ERC721) {
+            IERC721Upgradeable(collateralInfo._collateralAddress)
+            .safeTransferFrom(
+                borrower,
+                address(this),
+                collateralInfo._tokenId
+            );
+            collateralEscrow.depositAsset(
+                ICollateralEscrowV1.CollateralType.ERC721,
+                collateralInfo._collateralAddress,
+                collateralInfo._amount,
+                collateralInfo._tokenId
+            );
+        }
+        if (collateralInfo._collateralType == ICollateralEscrowV1.CollateralType.ERC1155) {
+            bytes memory data;
+            IERC1155Upgradeable(collateralInfo._collateralAddress)
+                .safeTransferFrom(
+                    borrower,
+                    address(this),
+                    collateralInfo._tokenId,
+                    collateralInfo._amount,
+                    data
+                );
+            collateralEscrow.depositAsset(
+                ICollateralEscrowV1.CollateralType.ERC1155,
+                collateralInfo._collateralAddress,
+                collateralInfo._amount,
+                collateralInfo._tokenId
+            );
+        }
+    }
+
+    /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     */
+    function withdraw(uint256 _bidId) external {
+        if (_isBidCollateralBacked[_bidId]) {
+            BidState bidState = TellerV2(owner()).getBidState(_bidId);
+            require(
```

Comment:
I dont love the use of > here because it seems sloppy.  If the order of ENUM were ever changed in the future this would cause a massive hack/vulnerability.   We should use equality like ==PAID and ==LIQUIDATED to be safe.